Increase bar plot stroke width

jayeclark opened the pull request:
Signed-off-by: Jay Clark <jay@jayeclark.dev>

## Description

Increases stroke width on highlighted city in bar chart, changes opacity to further increase visibility.

Closes #31 

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Simple d3 style attribute change, result verified visually.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

## Next Steps
- [x] Assign a reviewer based on the [code owner document](https://github.com/jamie-sgro/dashboard/blob/develop/.github/CODEOWNERS).

- [ ] Once your review is approved, merge and delete the feature branch

On behalf of the Cities Index Dev Team, thank you for your hard work! ✨

jamie-sgro reviewed (approved):
This looks quite nice. I like the touch about almost adding a spotlight by dimming the rest of the bars when selected. 

What are your thoughts on swapping the "0.9" magic number to a constant that we could declare at the top?

I'll approve regardless, but feel free to add the `const` if you feel compelled.

Thanks again!

jayeclark commented:
@jamie-sgro That makes a lot of sense to me. Putting it at the top will make it much easier to adjust in the future if a random person is working on the repository.  

Actually, right now it doesn't *quite* have the functionality you described -- the bars in that graph are _always_ at 0.9 opacity and then once a bar is highlighted it has 100% opacity. But what you described seems much better (that the bars only go to 0.9 once a city is selected, by default before a city is selected all bars should all be at 100% opacity.) 

I can put in an additional PR tomorrow that moves that 0.9 number to a const at the top of the page, and improves the opacity toggle functionality (signing off for the day soon today.)

jamie-sgro commented:
No rush! And I totally get where you're coming from. What I could see as a neat feature in the future would be the ability to **deselect** a city and have the stroke set to zero, and all bars set back to 100% opacity, but I'd have to look into whether the radio buttons currently being used can easily be toggled on and off (I believe we can only select a new value, but I could be mistaken)

jayeclark commented:
@jamie-sgro I moved the opacity to the class constructor and cleaned up a few more things (removed an unnecessary attribute setting that I had added, changed the opacity to be at the halfway point between 1 and the onHover highlight value of 0.7, updated the tooltips so that if the data name & description are the same, only the name will show - so there are no longer tooltips that say Halifax: Halifax.)

Toggling radio buttons on & off will require some work/exploration (there is not an easy way, they don't even register as `checked=true` when they are selected), but I'll take a stab at it in the coming weeks! 

As a user, I expected to be able to click on a particular bar in the cities bar plot and be able to view the data for that city in the chart to the left, so it might be nice to add that feature at the same time as well (as an alternative to selecting the city using the radio buttons.)

jamie-sgro commented on src/barplot.ts:
```diff
@@ -345,7 +349,7 @@ export class Barplot {
   onMouseover(data: DataPoint, index: number) {
     assertType(this, Barplot);
-    const initialText = data.name + ": " + data.description;
+    const initialText = data.name === data.description ? data.name :  data.name + ": " + data.description;
```
That's awesome, I love that your implementation still ensures that the first `Barplot()` retains it's original functionality while supporting some finer tuning on the second `Barplot()`. The ternary operator is a nice touch as well!

jamie-sgro reviewed (approved):
Approved, I've listed responses to your notes below:

>As a user, I expected to be able to click on a particular bar in the cities bar plot and be able to view the data for that city in the chart to the left, so it might be nice to add that feature at the same time as well (as an alternative to selecting the city using the radio buttons.)

I agree, especially since when you hover over the bar it flashes. Back in the day we had an `onClick()` method for the barplots that would trigger an update on a corresponding leaflet map, but when municipalities in Canada found it too confusing, we dropped in and reduced it's functionality. It'd be something I'd love to see brought back into the dashboard!

I'd encourage you to make an issue for the `Barplot.onClick()` functionality at the very least (and an issue for what your were mentioning for the toggle of radios on/off) so we can keep tabs on it. No pressure on implementing the same, but I've really appreciated your contributions to this project. I believe you could make a meaningful impact overall 😄
